flake.nix

```diff
@@ -17,22 +20,67 @@
     };
   };
 
-  outputs = { self, nixpkgs, flake-utils, cardano-node }:
+  outputs = { self, nixpkgs, flake-utils, cardano-node, poetry2nix }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
           pkgs = nixpkgs.legacyPackages.${system};
-          python3 = pkgs.python3.override {
-            packageOverrides = self: _: {
-              allure = self.callPackage ./nix/allure.nix { };
-              pytest-allure = self.callPackage ./nix/pytest-allure.nix { };
-              pytest-select = self.callPackage ./nix/pytest-select.nix {};
-              cardano-clusterlib = self.callPackage ./nix/cardano-clusterlib.nix { };
-            };
+          p2n = poetry2nix.legacyPackages.${system};
```

Comment:
```suggestion
          # As we are using [poetry](...) as the python dependency manager for cardano-node-tests, we will be using
          # [poetry2nix](https://github.com/nix-community/poetry2nix) to convert the poetry project (pyproject.toml,
          # and poetry.lock) into a nix-buildable expression. This is preferable over using `pkgs.python3.withPackages`
          # as it adheres to the poetry setup instead of replicating it in nix again.
          
          p2n = poetry2nix.legacyPackages.${system};
```